packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -0,0 +1,243 @@
+        const attachMessage = message.contents;
+        // The local object has already been attached
+        if (local) {
+            assert(this.pendingAttach.has(attachMessage.id));
+            this.contexts.get(attachMessage.id)?.emit("attached");
+            this.pendingAttach.delete(attachMessage.id);
+            return;
+        }
+
+         // If a non-local operation then go and create the object, otherwise mark it as officially attached.
+        if (this.contexts.has(attachMessage.id)) {
+            const error = new Error("DataCorruption: Duplicate data store created with existing ID");
+            this.logger.sendErrorEvent({
+                eventName: "DuplicateDataStoreId",
+                sequenceNumber: message.sequenceNumber,
+                clientId: message.clientId,
+                referenceSequenceNumber: message.referenceSequenceNumber,
+            }, error);
+            throw error;
+        }
+
+        const flatBlobs = new Map<string, string>();
+        let flatBlobsP = Promise.resolve(flatBlobs);
+        let snapshotTreeP: Promise<ISnapshotTree> | null = null;
+        if (attachMessage.snapshot) {
+            snapshotTreeP = buildSnapshotTree(attachMessage.snapshot.entries, flatBlobs);
+            // flatBlobs' validity is contingent on snapshotTreeP's resolution
+            flatBlobsP = snapshotTreeP.then(() => { return flatBlobs; });
+        }
+
+        // Include the type of attach message which is the pkg of the store to be
+        // used by RemotedFluidDataStoreContext in case it is not in the snapshot.
+        const pkg = [attachMessage.type];
+        const remotedFluidDataStoreContext = new RemotedFluidDataStoreContext(
+            attachMessage.id,
+            snapshotTreeP,
+            this.runtime,
+            new BlobCacheStorageService(this.runtime.storage, flatBlobsP),
+            this.runtime.scope,
+            this.summaryTracker.createOrGetChild(attachMessage.id, message.sequenceNumber),
+            this.getCreateChildSummarizerNodeFn(
+                attachMessage.id,
+                {
+                    type: CreateSummarizerNodeSource.FromAttach,
+                    sequenceNumber: message.sequenceNumber,
+                    snapshot: attachMessage.snapshot ?? {
+                        id: null,
+                        entries: [createAttributesBlob(pkg, true /* isRootDataStore */)],
+                    },
+                }),
+            pkg);
+
+        // Resolve pending gets and store off any new ones
+       this.contexts.addBoundOrRemote(attachMessage.id, remotedFluidDataStoreContext);
+
+        // Equivalent of nextTick() - Prefetch once all current ops have completed
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        Promise.resolve().then(async () => remotedFluidDataStoreContext.realize());
+    }
+
+    public  bindFluidDataStore(fluidDataStoreRuntime: IFluidDataStoreChannel): void {
```

Comment:
Add a disposed check here, to cover for the removal of `this.verifyNotClose()`?

Reply:
i ran into issues with that check here, as the current dispose does some async stuff. since this class is internal to the containerRuntime, which does the check, we shouldn't need it, but i do think we need cleanup in this area

Reply:
Well containerRuntime isn't doing the check anymore, unless I missed it.  It's probably fine, I imagine something will immediately blow up if things have been disposed...